### ISSUE TITLE
thinking_budget is silently ignored on models using native reasoning chat templates

### ISSUE THREAD

**Issue by xunlinkx** [jundot/omlx#3112]
## Summary

When I send requests with `thinking_budget` (or configure `thinking_budget_enabled: true` in `settings.json`) for models that format reasoning natively via Jinja (e.g. Qwen 3.5/3.6/3.8, FableVibes, DeepSeek), the scheduler silently skips attaching `ThinkingBudgetProcessor`. The model reasons unconstrained and burns the entire `max_tokens` allocation (e.g. hitting `finish_reason=length` after 8192 tokens), truncating mid-stream tool calls and stalling our agent loops.

## Environment

- Apple Silicon (M3 / M4 Max 48 GB), macOS 26.5.2
- oMLX 0.6.3rc2 (HEAD-fa3e94b), MLX 0.32.0
- Models: `Qwen3.6-14B-A3B-FableVibes-oQ4e`, `Qwen3.8-27B-oQ4e-mtp`, and other native reasoning templates where `<think>\n` is rendered directly by Jinja.

## What happens / Root cause

In `omlx/scheduler.py` (`_build_sampler_and_processors`):

```python
if (
    sampling_params.thinking_budget is not None
    and request is not None
    and (
        getattr(request, "needs_think_prefix", False)
        or self._get_output_parser_thinking_end_text() is not None
    )
):
    think_end_ids = self._resolve_think_end_token_ids()
    if think_end_ids:
        ...
        logits_processors.append(processor)
```

1. When a model's Jinja template handles thinking natively (such as Qwen/FableVibes appending `<think>\n` when `add_generation_prompt=True`), the prompt already contains the opener, so `_detect_needs_think_prefix(request)` returns `False`.
2. Because `request.needs_think_prefix` is `False` and `_get_output_parser_thinking_end_text()` is `None`, the scheduler skips `ThinkingBudgetProcessor` construction entirely.
3. The request's `thinking_budget` is dropped on the decode path, allowing runaway reasoning to consume the entire token budget.

## Proposed fix

Gate `ThinkingBudgetProcessor` construction directly on `sampling_params.thinking_budget is not None and request is not None`. Tokenizer/parser think-token resolution (`self._resolve_think_end_token_ids()`) already returns `None` for non-reasoning models, safely preventing unintended attachment on standard models. Dedicated output parsers (Gemma 4, Harmony, MiniMax) and non-reasoning tokenizers remain protected by their existing resolution paths.

**Comment by beaglemoo** [comment]
Datapoint from an M5 Pro on the official 0.6.3rc2 DMG with stock checkpoints, where the budget does apply: Qwen3.6-35B-A3B via a model-settings budget of 256 -> 271 completion tokens vs 568 unbounded; Qwen3.8-27B with a request-level `thinking_budget: 256` -> reasoning cut from 3,338 chars / 1,832 tokens to 814 / 678, clean `finish_reason=stop` both ways. So the skip isn't universal to Qwen native-reasoning templates — on these checkpoints the gate passes (either `needs_think_prefix` or the parser's thinking-end text resolves). It may be specific to your conversion's chat template — does its rendered prompt end with `<think>\n` when `add_generation_prompt=true`? Your proposed gate widening still looks reasonable as robustness since `_resolve_think_end_token_ids()` fails closed, but the affected-model list is narrower than titled.
